tokio/src/net/tcp/split.rs

```diff
@@ -123,7 +131,7 @@ impl ReadHalf<'_> {
     ///
     /// [`read`]: fn@crate::io::AsyncReadExt::read
     /// [`AsyncReadExt`]: trait@crate::io::AsyncReadExt
-    pub async fn peek(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+    pub async fn peek(&mut self, buf: &mut ReadBuf<'_>) -> io::Result<usize> {
```

Comment:
We shouldn't change this one for the same reasons that `AsyncReadExt::read` also takes `&mut [u8]`.